crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4571,9 +4576,12 @@ impl<'db> TypeInferenceBuilder<'db> {
         } = subscript;
 
         match value_ty {
-            Type::KnownInstance(known_instance) => {
-                self.infer_parameterized_known_instance_type_expression(known_instance, slice)
-            }
+            Type::KnownInstance(known_instance) => self
+                .infer_parameterized_known_instance_type_expression(
+                    subscript,
+                    known_instance,
+                    slice,
+                ),
```

Comment:
nit: I think we can skip passing the `slice` if we pass the `subscript` as it can be fetched from the subscript directly